package/src/main/java/com/microsoft/azure/cosmos/cassandra/CosmosRetryPolicy.java

```diff
@@ -117,49 +127,34 @@ public RetryDecision onWriteTimeout(
             int requiredAcks,
             int receivedAcks,
             int retryNumber) {
-
         return retryManyTimesOrThrow(retryNumber);
     }
-    
+
     private final static Random random = new Random();
     private final int growingBackOffSaltMillis = 2000;
     private final int fixedBackOffTimeMillis;
     private final int growingBackOffTimeMillis;
     private final int maxRetryCount;
+    private boolean useRetryMillisIfAvailable;
 
     private RetryDecision retryManyTimesOrThrow(int retryNumber) {
-
-        RetryDecision retryDecision;
-
-        if (this.maxRetryCount == -1) {
-            retryDecision = RetryDecision.retry(null);
-        } else {
-            if (retryNumber < this.maxRetryCount) {
-                retryDecision = RetryDecision.retry(null);
-            } else {
-                retryDecision = RetryDecision.rethrow();
-            }
-        }
-
-        return retryDecision;
+        return (this.maxRetryCount == -1 || retryNumber < this.maxRetryCount) ?
+                RetryDecision.retry(null) : RetryDecision.rethrow();
     }
 
-    private RetryDecision retryManyTimesWithBackOffOrThrow(int retryNumber) throws InterruptedException {
+    public int getRetryAfterMs(String exceptionString){
```

Comment:
Can you add a comment here describing what a sample exception message would look like?